pyvisa_py/protocols/hislip.py

```diff
@@ -250,463 +567,179 @@ def initialize(self, version=(1, 0), vendor_id=b"xx", sub_address=b"hislip0"):
 
-    def async_lock_request(self, timeout, lock_string=""):
+    def async_lock_request(self, timeout: float, lock_string: str = "") -> str:
         """
         perform an AsyncLock request transaction.
         returns the lock_response from the AsyncLockResponse packet.
         """
         # async_lock transaction:
         #     C->S: AsyncLock
         #     S->C: AsyncLockResponse
-        header = struct.pack(
-            "!2sBBIQ",
-            b"HS",
-            MESSAGETYPE["AsyncLock"],
-            1,
-            1e3 * timeout,
-            len(lock_string),
-        )
-        self._async.sendall(header + lock_string)
-        response_header = receive_header(
-            self._async, expected_message_type="AsyncLockResponse"
-        )
-        return response_header.lock_response
+        ctrl_code = LOCKCONTROLCODE["request"]
+        timeout_ms = int(1e3 * timeout)
+        send_msg(self._async, "AsyncLock", ctrl_code, timeout_ms, lock_string.encode())
+        response = AsyncLockResponse(self._async)
+        return response.lock_response
 
-    def async_lock_release(self):
+    def async_lock_release(self) -> str:
         """
         perform an AsyncLock release transaction.
         returns the lock_response from the AsyncLockResponse packet.
         """
         # async_lock transaction:
         #     C->S: AsyncLock
         #     S->C: AsyncLockResponse
-        header = struct.pack(
-            "!2sBBIQ", b"HS", MESSAGETYPE["AsyncLock"], 0, self.last_message_id, 0
-        )
-        self._async.sendall(header)
-        response_header = receive_header(
-            self._async, expected_message_type="AsyncLockResponse"
-        )
-        return response_header.lock_response
+        ctrl_code = LOCKCONTROLCODE["release"]
+        send_msg(self._async, "AsyncLock", ctrl_code, self._last_message_id)
+        response = AsyncLockResponse(self._async)
+        return response.lock_response
 
-    def async_remote_local_control(self, remotelocalcontrol):
+    def async_remote_local_control(self, remotelocalcontrol: str) -> None:
         """
         perform an AsyncRemoteLocalControl transaction.
         """
         # remote_local transaction:
         #     C->S: AsyncRemoteLocalControl
         #     S->C: AsyncRemoteLocalResponse
-        header = struct.pack(
-            "!2sBBIQ",
-            b"HS",
-            MESSAGETYPE["AsyncRemoteLocalControl"],
-            REMOTELOCALCONTROLCODE[remotelocalcontrol],
-            self.last_message_id,
-            0,
-        )
-        self._async.sendall(header)
-        response_header = receive_header(
-            self._async, expected_message_type="AsyncRemoteLocalResponse"
+        ctrl_code = REMOTELOCALCONTROLCODE[remotelocalcontrol]
+        send_msg(
+            self._async, "AsyncRemoteLocalControl", ctrl_code, self._last_message_id
         )
-        return response_header.server_status
+        _ = AsyncRemoteLocalResponse(self._async)
```

Comment:
```suggestion
       AsyncRemoteLocalResponse(self._async)
```

Reply:
I've added this change into my latest commit.